core/DataAccess/ArchiveSelector.php

```diff
@@ -623,7 +623,7 @@ private static function getSqlTemplateToFetchArchiveData(array $recordNames, $id
                                     AND SUBSTRING(name, $nameEndAppendix, 1) <= '9')";
 
             $whereNameIs = "(name = ? OR (name LIKE ? AND ( $checkForChunkBlob OR $checkForSubtableId ) ))";
-            $bind = array($name, $name . '%');
+            $bind = array($name, str_replace("_", "\_", $name) . '%');
```

Comment:
Should we also escape `%` character? Not sure if it can appear in the archive name, but it has a special meaning in the LIKE clause, so perhaps it's cleaner?

```suggestion
            $bind = array($name, addcslashes($name, '%_') . '%');
```

https://3v4l.org/89ZfB

Reply:
Archive names are restricted to be alphanumeric including `_` and `-`. See https://github.com/matomo-org/matomo/blob/0545624d96956218ce881d960ae87a9820cdb56d/core/ArchiveProcessor/Record.php#L92-L100
So this wouldn't be needed. But it also won't hurt...